tensorflow_io/core/python/experimental/avro_dataset_ops.py

```diff
@@ -0,0 +1,533 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# From CSV dataset
+# https://github.com/tensorflow/tensorflow/blob/v2.0.0/tensorflow/python/data/experimental/ops/readers.py
+
+# Parse example dataset
+# https://github.com/tensorflow/tensorflow/blob/v2.0.0/tensorflow/python/data/experimental/ops/parsing_ops.py
+
+# For tf export use examples see
+# sql_dataset_test_base.py for use and readers.py for definition
+
+import collections
+import re
+
+import tensorflow as tf
+from tensorflow.python.data.util import structure
+from tensorflow.python.ops import parsing_ops
+from tensorflow.python.data.ops.dataset_ops import DatasetSource
+from tensorflow.python.data.experimental.ops import readers
+from tensorflow_io.core.python.ops import core_ops
+
+# Note: I've hidden the dataset because it does not apply the mapping for
+# sparse tensors
+# Such mapping is not possible inside the dataset, rather it needs to happen
+# through a map on the output of the dataset which is a map of keys to tensors
+# This can be changed when eager mode is the default and the only mode supported
+class _AvroDataset(DatasetSource):
```

Comment:
There is nothing special about `DatasetSource`:
https://github.com/tensorflow/tensorflow/blob/9239c61f201af6e8147a8ab7330edd87fea5a6ad/tensorflow/python/data/ops/dataset_ops.py#L2299-L2303

Can you:
1. use `_AvroDataset(tf.data.Dataset)`
2. add the following to the class (in fact it might not even be needed)?
   ```
   def _inputs(self):
       return []
   ```

Reply:
updated.